src/Stheno.jl

```diff
@@ -60,8 +61,8 @@ module Stheno
 
     function __init__()
         @require Flux="587475ba-b771-5e3f-ad9e-33799f191a9c" begin
-            include(joinpath("gp", "add_functor.jl"))
-            include(joinpath("gp", "neural_kernel_network.jl"))
+            include(joinpath("flux", "add_functor.jl"))
+            include(joinpath("flux", "neural_kernel_network.jl"))
```

Comment:
Since neural kernel network can be viewed just as another kernel function, maybe it's ok to leave it in `gp`. Maybe it's good to extract `LinearLayer` & `product` functions into a new file and put it in the `flux` folder, and leave `neural_kernel_network.jl` in `gp`. What do think ?

Reply:
The reason for the move is to properly separate out the functionality. The NKN stuff doesn't work in a meaningful sense unless you load Flux because you don't have access to the layers, so it makes sense to me to only load it _if_ you load Flux.